camel/societies/workforce/workforce.py

```diff
@@ -342,8 +349,8 @@ def _create_new_agent(self, role: str, sys_msg: str) -> ChatAgent:
             content=sys_msg,
         )
 
-        if self.new_worker_agent_kwargs is not None:
-            return ChatAgent(worker_sys_msg, **self.new_worker_agent_kwargs)
+        if self.new_worker_agent is None:
+            return ChatAgent(worker_sys_msg)
```

Comment:
This part confuses me. A new worker agent should be created inside workforce. Maybe we should allow user to setup `model` and specify the available tools. The available tools can be used by a tool selection agent to select tools for the newly created agent. What do you think?